aws-sagemaker-hosted-scorer/src/main/java/ai/h2o/mojos/deploy/sagemaker/hosted/Serve.java

```diff
@@ -0,0 +1,103 @@
+package ai.h2o.mojos.deploy.sagemaker.hosted;
+
+import ai.h2o.mojos.runtime.utils.SimpleCSV;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import ai.h2o.mojos.runtime.MojoPipeline;
+import ai.h2o.mojos.runtime.frame.MojoFrame;
+import ai.h2o.mojos.runtime.frame.MojoFrameBuilder;
+import ai.h2o.mojos.runtime.frame.MojoRowBuilder;
+
+import java.io.File;
+import java.util.HashMap;
+
+@Component
+public class Serve {
+    // Used to prevent REST API requests before the server is initialized and ready to handle them.
+    private volatile boolean ready = false;
+
+    private void setReady() {
+        ready = true;
+    }
+
+    boolean isReady() {
+        return ready;
+    }
+
+    private volatile MojoPipeline model;
+
+    @Autowired
+    public Serve() {
+    }
+
+    /**
+     * Call this immediately after the constructor above.
+     * Since this object is used as an Autowired Spring object, the constructor call is not actually visible.
+     */
+    void init() throws Exception {
+        String[] locationsToCheck = {
+                "/opt/ml/model/pipeline.mojo",
+                "./pipeline.mojo"
+        };
+
+        File f = null;
+        for (String location : locationsToCheck) {
+            File tmp = new File(location);
+            if (tmp.exists()) {
+                f = tmp;
+                break;
+            }
+        }
+
+        if (f == null) {
+            System.out.println("ERROR: pipeline.mojo file not found");
+            System.exit(1);
+        }
+
+        long startMillis = System.currentTimeMillis();
+        model = MojoPipeline.loadFrom(f.getAbsolutePath());
+        long endMillis = System.currentTimeMillis();
+        long deltaMillis = endMillis - startMillis;
+        System.out.println("INFO: MojoPipeline.loadFrom took " + deltaMillis + " millis");
+    }
+
+    HashMap<String,String> invocations(HashMap<String,String> inputs) {
```

Comment:
I think we already have most of this code in the shared libraries. For example the LocalRestServer does not do much work itself: https://github.com/h2oai/dai-deployment-templates/blob/master/local-rest-scorer/src/main/java/ai/h2o/mojos/deploy/local/rest/controller/ModelsApiController.java.

Would it make sense to take a similar approach? The benefit I see is that we would have the same REST interface for free (at least the payloads) and have the request transformation already covered by tests. Wdyt?